scenario-compiler/src/main/java/pl/touk/nussknacker/engine/spel/parser/NuSpelExpressionParser.java

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pl.touk.nussknacker.engine.spel.parser;
+
+import org.springframework.expression.Expression;
+import org.springframework.expression.ParseException;
+import org.springframework.expression.ParserContext;
+import org.springframework.expression.common.CompositeStringExpression;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.expression.spel.SpelParserConfiguration;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.lang.Nullable;
+import pl.touk.nussknacker.engine.expression.PositionRange;
+
+import java.util.*;
+
+/**
+ * ATTENTION
+ * This class is a copy-paste of TemplateAwareExpressionParser from Spring with some slight modification - to also return position of parsed expressions.
+ * Start and end positions are needed for error locations in template expressions and for expression suggestions - see SpelExpressionSuggester class.
+ * /
+ * <p>
+ * /**
+ * An expression parser that understands templates. It can be subclassed by expression
+ * parsers that do not offer first class support for templating.
+ *
+ * @author Keith Donald
+ * @author Juergen Hoeller
+ * @author Andy Clement
+ * @since 3.0
+ */
+public class NuSpelExpressionParser {
```

Comment:
This class was already in the repo (`NuTemplateAwareExpressionParser`). The suggestion mechanism used it. I had to make it used also during expression parsing in order to provide information about location of an expression inside the template which is used in `PositionRange.shiftRight`